File: 6/6-7/Java/RTO.java
import java.lang.Math;

public class RTO {
	public static void main(String[] args) {
		double RTTArray[] = new double[] { 26, 32, 24, 26, 26, 28, 26, 26, 28, 26, 26, 28, 30 };
		double Alpha = 0.125, Beita = 0.25;
		
		double RTT, RTTs, RTTd, RTO;
		RTT = RTTArray[0];
		RTTs = RTT;
		RTTd = RTT / 2;
		RTO = RTTs + 4 * RTTd;
		System.out.printf("Initial RTT: %f\n", RTT);
		
		for (int i = 1; i < RTTArray.length; i++) {
			RTT = RTTArray[i];
			RTTs = (1 - Alpha) * RTTs + Alpha * RTT;
			RTTd = (1 - Beita) * RTTd + Beita * Math.abs(RTTs - RTT);
			RTO = RTTs + 4 * RTTd;
			System.out.printf("Round: %d, RTT: %f, RTTs: %f, RTO: %f\n", i, RTT, RTTs, RTO);
		}
	}
}

File: 6/6-7/Python/RTO.py
import os

if __name__ == '__main__':
    RTTArray = [26, 32, 24, 26, 26, 28, 26, 26, 28, 26, 26, 28, 30]
    Alpha = 0.125
    Beita = 0.25
    RTT = RTTArray[0]
    RTTs = RTT
    RTTd = RTT / 2
    RTO = RTTs + 4 * RTTd
    print("Initial RTT: {}".format(RTT))

    for i in range(1, len(RTTArray)):
        RTT = RTTArray[i]
        RTTs = (1 - Alpha) * RTTs + Alpha * RTT
        RTTd = (1 - Beita) * RTTd + Beita * abs(RTTs - RTT)
        RTO = RTTs + 4 * RTTd
        print("Round: {}, RTT: {}, RTTs: {}, RTO: {}".format(i, RTT, RTTs, RTO))
    os.system("pause")

